kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyBackendHandler.java

```diff
@@ -22,10 +29,13 @@ public class KafkaProxyBackendHandler extends ChannelInboundHandlerAdapter {
     private final ChannelHandlerContext inboundCtx;
     private ChannelHandlerContext blockedOutboundCtx;
     private boolean unflushedWrites;
+    private final Map<Class<? extends Exception>, Function<Throwable, ?>> responsesByExceptionType;
 
     public KafkaProxyBackendHandler(KafkaProxyFrontendHandler frontendHandler, ChannelHandlerContext inboundCtx) {
         this.frontendHandler = frontendHandler;
         this.inboundCtx = requireNonNull(inboundCtx);
+        responsesByExceptionType = new ConcurrentHashMap<>();
+        responsesByExceptionType.put(SSLHandshakeException.class, UnknownServerException::new);
```

Comment:
this doesn't actually do what I want. the client still sees a timeout and not the `UnknownServerException`

Reply:
is your intent to organise for the client's first RPC to get a response containing the USE in this case?  Maybe that's worthy of a separate PR.  Maybe we'd choose not to land that for 0.7.1

Reply:
Yeah. I think thats probably the right call. I'll move the commits shortly

Reply:
Split out as https://github.com/kroxylicious/kroxylicious/pull/1475